Support RFP=1 in Linux Consumption for V3.x.

khkh-ms opened the pull request:
# This PR is a same as the original PR at https://github.com/Azure/azure-functions-host/pull/8374 but for V3 branch. 

<!-- Please provide all the information below.  -->

### Issue describing the changes in this PR
Currently WEB_RUN_FROM_PACKAGE=1 is not supported by Linux Consumption. This change adds that support.  The change has corresponding KUDU change.  The commits for this PR were cherry-picked from [original PR here](https://github.com/Azure/azure-functions-host/pull/8374) for dev branch. 

resolves #issue_for_this_pr

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [ ] I have added all required tests (Unit tests, E2E tests)

<!-- Optional: delete if not applicable  -->
### Additional information

Additional PR information

khkh-ms committed:
lin con rfp=1

khkh-ms committed:
minor change

khkh-ms committed:
fixed an issue

khkh-ms committed:
logs for testings

khkh-ms committed:
test without metrics

khkh-ms committed:
change in getpackage type

khkh-ms committed:
cleanup.

khkh-ms committed:
Unit test fix.

khkh-ms committed:
Added unit tests

khkh-ms committed:
Removed the code witch is not needed.

khkh-ms committed:
Fix after testing.

khkh-ms committed:
Addressed the review comments.

# Conflicts:
#	src/WebJobs.Script.WebHost/Management/LinuxSpecialization/PackageDownloadHandler.cs
#	test/WebJobs.Script.Tests.Integration/Management/PackageDownloadHandlerTests.cs
#	test/WebJobs.Script.Tests.Integration/Management/RunFromPackageHandlerTests.cs

khkh-ms committed:
Passing FileSystem object.

# Conflicts:
#	test/WebJobs.Script.Tests.Integration/Management/RunFromPackageHandlerTests.cs

khkh-ms committed:
Added unit tests

# Conflicts:
#	test/WebJobs.Script.Tests.Integration/Management/PackageDownloadHandlerTests.cs

khkh-ms committed:
typos.

khkh-ms committed:
Addressed the review comments.

khkh-ms committed:
Fixed the merge issue.s

khkh-ms committed:
merge fix

fabiocav commented:
@khkh-ms can you please share the details/motivation for this change so we have the right context for a review? thanks!

khkh-ms commented:
@fabiocav I am working to support RFP=1 on Linux Consumption which is not supported at the moment.  The commits for this PR were cherry-picked from [original PR here](https://github.com/Azure/azure-functions-host/pull/8374) for dev branch. 

There is a KuduLite change for the same which was merged last week.

khkh-ms committed:
Feedback

# Conflicts:
#	src/WebJobs.Script.WebHost/Management/LinuxSpecialization/PackageDownloadHandler.cs
#	test/WebJobs.Script.Tests.Integration/Management/PackageDownloadHandlerTests.cs
#	test/WebJobs.Script.Tests.Integration/Management/RunFromPackageHandlerTests.cs

khkh-ms committed:
rfp=1 is now Squashfs

khkh-ms committed:
minor fix for v3.

khkh-ms commented:
Hey @balag0 , you reviewed the original PR for v4 at https://github.com/Azure/azure-functions-host/pull/8374. Please approve this one. This is same PR for V3 branch. I have cherry-picked the commits from the original PR.
